src/status_im/mailserver/core.cljs

```diff
@@ -446,23 +453,32 @@
         (fx/merge cofx
                   {:db (dissoc db :mailserver/current-request)}
                   (process-next-messages-request))
-        (fx/merge cofx
-                  {:db (-> db
-                           (dissoc :mailserver/current-request)
-                           (update :mailserver/topics merge mailserver-topics))
-                   :data-store/tx (mapv (fn [[topic mailserver-topic]]
-                                          (data-store.mailservers/save-mailserver-topic-tx
-                                           {:topic topic
-                                            :mailserver-topic mailserver-topic}))
-                                        mailserver-topics)}
-                  (process-next-messages-request))))))
+        ;; If a cursor is returned, add cursor and fire request again
+        (if (seq cursor)
+          (when-let [mailserver (get-mailserver-when-ready cofx)]
+
+            {:mailserver/request-messages {:web3 (:web3 db)
+                                           :mailserver    mailserver
+                                           :request (assoc request :cursor cursor)}})
+          (fx/merge cofx
+                    {:db (-> db
+                             (dissoc :mailserver/current-request)
+                             (update :mailserver/topics merge mailserver-topics))
+                     :data-store/tx (mapv (fn [[topic mailserver-topic]]
+                                            (data-store.mailservers/save-mailserver-topic-tx
+                                             {:topic topic
+                                              :mailserver-topic mailserver-topic}))
+                                          mailserver-topics)}
+                    (process-next-messages-request)))))))
 
 (fx/defn retry-next-messages-request
   [{:keys [db] :as cofx}]
   (fx/merge cofx
             {:db (dissoc db :mailserver/request-error)}
             (process-next-messages-request)))
 
+;; At some point we should update `last-request`, as eventually we want to move
+;; on, rather then keep asking for the same data, say after n amounts of attempts
```

Comment:
I'm not sure if that commit is the right place, but there is this issue based on the previous postmortem: https://github.com/status-im/status-react/issues/7083

Reply:
Yes, I have added cursor to this PR as adding a request last  24 hours makes it easier to test, the commit partially solves the issue (we fetch much less data with each request , max 200 messages), but it's still a all-or-nothing approach (if for whatever reason a request fails it will start from scratch), that I can take care separately, I want to validate cursors are working as expected first.